test/test_dtd.py

```diff
@@ -58,6 +60,42 @@ def test_invalid_inputs(self):
                     self.dtd.validate(parsed), f"{name} validates but shouldn't"
                 )
 
+    def test_backwards_compatibility(self):
+        # the DTD needs to be backwards compatible as long as the major version does not change
+        versions = ["ras-dtd-1.0.readalong", "ras-dtd-1.1.readalong"]
+        for name in versions:
+            path = os.path.join(dirname(__file__), "data", name.strip())
+            # DTD is text, XML is binary... okay
+            with open(path, "rb") as infh:
+                try:
+                    parsed = load_xml(infh)
+                    self.assertTrue(
+                        self.dtd.validate(parsed), f"{name} does not validate"
+                    )
+                except etree.ParseError as e:
+                    self.fail("Error parsing XML input file %s: %s." % (path, e))
```

Comment:
That's nice and clean, making things explicit, but generally, unittest handles this for us: every uncaught exception is automatically counted as a test failure by unittest.

Reply:
I hope this will help debugging (especially in the CI).